jenkinsci/jenkins
# Add search bar for top level settings in Manage Jenkins

https://user-images.githubusercontent.com/43062514/198851674-054336f6-0f6f-4473-990f-131f53669513.mov

This PR adds a small search bar which searches the top level (just visible) settings on the Manage Jenkins page, it's pretty simple in it's featureset but it makes Jenkins a little easier to navigate via keyboard.

It also improves the search bar component, it's now possible to provide it a list of items and the search bar will provide suggestions from that set of data - this will be added to the Design Library.

It would be great to have this extended in future to include items from each settings page, but for now I felt that making it more keyboard accessible + improving the search bar component was enough.

---

<img width="284" alt="image" src="https://user-images.githubusercontent.com/43062514/198856142-18b1bfe6-f3cd-4365-a2ec-157abd0d5a07.png">

One of the things noticed whilst building this is the unnecessary prefixes, e.g. '_Configure System', '_Manage_ Plugins', '_Manage_ Nodes and Clouds' - you're already in Manage Jenkins so it seems unnecessary to include the word 'Configure/Manage' in many of the items. I'd propose dropping the word from each item in the future + this would make search results easier to read, any thoughts?

### Testing done

* Open up Manage Jenkins, search should autofocus, type to filter items on the page. The only item that won't appear is 'Reload configuration from disk' as this is an action, not a link.
* Use arrow keys (up/down) to focus different items, press enter key to navigate to the currently selected one.
* Open up a page with the search bar e.g. a build, plugin manager (not Manage Jenkins) and check existing functionality still works, check console to see that there are no errors

### Proposed changelog entries

- Add search bar for top level settings in Manage Jenkins.

### Proposed upgrade guidelines

N/A

### Developer changelog

- The search bar component now displays search suggestions when passed a dataset (see Jenkins Design Library)

### Submitter checklist

  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during upgrade.

### Desired reviewers

@jenkinsci/sig-ux 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`:



<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/7314"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
